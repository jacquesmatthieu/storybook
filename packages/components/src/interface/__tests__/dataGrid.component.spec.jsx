import React from 'react';
import { shallow } from 'enzyme';

import { DataGrid, DataGridItem, DataGridRow } from '../dataGrid.component';

describe('DataGrid', () => {
  let props;

  beforeEach(() => {
    props = {
      columns: ['foo', 'bar', 'baz'],
      labels: ['hello', 'world'],
      rows: [42, 69],
    };
  });

  const getWrapper = () => shallow(<DataGrid {...props}>Foo</DataGrid>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DataGridRow', () => {
  let props;

  beforeEach(() => {
    props = {
      y: 42,
      h: 69,
    };
  });

  const getWrapper = () => shallow(<DataGridRow {...props}>Bar</DataGridRow>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DataGridItem', () => {
  let props;

  beforeEach(() => {
    props = {
      x: 1,
      w: 13,
      y: 42,
      h: 69,
    };
  });

  const getWrapper = () => shallow(<DataGridItem {...props}>Bar</DataGridItem>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
