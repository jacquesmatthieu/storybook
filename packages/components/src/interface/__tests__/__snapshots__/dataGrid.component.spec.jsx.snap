// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid should render properly 1`] = `
<dataGridcomponent__StyledDataGrid
  columns={
    Array [
      "foo",
      "bar",
      "baz",
    ]
  }
  rows={
    Array [
      42,
      69,
    ]
  }
>
  <dataGridcomponent__DataGridHeadItem
    h={1}
    key="hello"
    variant="tableHead"
    w={1}
    x={0}
    y={-1}
  >
    hello
  </dataGridcomponent__DataGridHeadItem>
  <dataGridcomponent__DataGridHeadItem
    h={1}
    key="world"
    variant="tableHead"
    w={1}
    x={1}
    y={-1}
  >
    world
  </dataGridcomponent__DataGridHeadItem>
  Foo
</dataGridcomponent__StyledDataGrid>
`;

exports[`DataGridItem should render properly 1`] = `
<dataGridcomponent__StyledDataGridItem
  h={69}
  variant="table"
  w={13}
  x={1}
  y={42}
>
  <tableRowcomponent__AntiOverflow>
    Bar
  </tableRowcomponent__AntiOverflow>
</dataGridcomponent__StyledDataGridItem>
`;

exports[`DataGridRow should render properly 1`] = `
<Fragment>
  <dataGridcomponent__StyledDataGridRow
    h={69}
    x={0}
    y={42}
  />
  Bar
  <dataGridcomponent__DataGridRowGap
    h={1}
    w={1}
    x={0}
    y={111}
  />
</Fragment>
`;
